test/driver.js

```diff
@@ -233,25 +254,7 @@ var rasterizeAnnotationLayer = (function rasterizeAnnotationLayerClosure() {
           pdfjsLib.AnnotationLayer.render(parameters);
 
           // Inline SVG images from text annotations.
-          var images = div.getElementsByTagName("img");
-          var imagePromises = inlineAnnotationImages(images);
-
-          await Promise.all(imagePromises).then(function (data) {
-            var loadedPromises = [];
-            for (var i = 0, ii = data.length; i < ii; i++) {
-              images[i].src = data[i];
-              loadedPromises.push(
-                new Promise(function (resolveImage, rejectImage) {
-                  images[i].onload = resolveImage;
-                  images[i].onerror = function (e) {
-                    rejectImage(new Error("Error loading image " + e));
-                  };
-                })
-              );
-            }
-            return loadedPromises;
-          });
-
+          await resolveImages(div);
```

Comment:
`await resolveImages(div)` is a list of promises, shouldn't you await them via `await Promise.all(await resolveImages(div))`?

Reply:
Or, you could add `await Promise.all(loadedPromises)` at the end of `resolveImages` and keep the code here as it is.

Reply:
The *second* suggestion is definitely the preferred way of doing things here!

---

Given that this patch only moves pre-existing code, it's actually quite interesting that old issue this hasn't caused any bugs (as far as we know).

Reply:
I locally got some errors because tiff images are not rendered by neither firefox nor chrome which is expected.
So I disabled promise rejection for image failures for the xfa reftests.